### ISSUE TITLE
Script generator: API for exporting scripts from the script generator

### ISSUE THREAD

**Issue by Tom-Willemsen** [ISISComputingGroup/IBEX#4167]
As a user or scientist, I need to be able to generate a Python script based on the contents of the script generator.

Acceptance criteria:
- As a user, I can create a python script which corresponds to the contents of the script generator.
- The API should be generic enough to support exporting to other file types in future, but these other file types are not part of this ticket
- Sending scripts directly to nicos is not part of this ticket

**Comment by DominicOram** [comment]
As discussed, I will review this after #4168 has been reviewed
